packages/nuxt/src/components/runtime/client-delayed-component.ts

```diff
@@ -0,0 +1,102 @@
+import {  defineAsyncComponent, defineComponent, getCurrentInstance, h, hydrateOnIdle, hydrateOnInteraction, hydrateOnMediaQuery, hydrateOnVisible, nextTick, onBeforeUnmount, onMounted, ref, watch } from 'vue'
+import type { AsyncComponentLoader, HydrationStrategy } from 'vue'
+import { onNuxtReady, useNuxtApp } from '#app'
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyIOComponent = (loader: AsyncComponentLoader) => {
+  return defineComponent({
+    inheritAttrs: false,
+    setup (_, { attrs }) {
+      if (import.meta.server) {
+        return () => h(defineAsyncComponent(loader), attrs)
+      }
+      const ready = ref(false)
+      const nuxt = useNuxtApp()
+      const instance = getCurrentInstance()!
+      onNuxtReady(() => ready.value = true)
+      // This is a hack to prevent hydration mismatches for all hydration strategies
+      return () => ready.value ? h(defineAsyncComponent({ loader, hydrate: hydrateOnVisible(attrs.hydrate as IntersectionObserverInit | undefined) })) : nuxt.isHydrating && instance.vnode.el ? h('div', attrs) : null

```

Comment:
While this hack prevents hydration mismatches for all types of hydrations, it also appears to ignore them altogether, and in stead it treats everything as a regular dynamic import (meaning it doesn't defer anything outside the initial client render)

Reply:
Marking as resolved for the time being due to the new solution. This could probably be explored a little further

Reply:
This is now extra resolved 👍